android-client-sdk/src/main/java/com/devcycle/sdk/android/api/EventQueue.kt

```diff
@@ -99,8 +103,13 @@ internal class EventQueue constructor(
             }
         }
 
-        if (eventQueue.size > 0 || !result.success) {
-            scheduler.scheduleWithDelay { run() }
+        if ((eventQueue.size > 0 || !result.success) && !isClosed.get()) {
+            scheduleJob = scheduler.scheduleWithDelay { run() }
+        }
+
+        // call the provided callback function if the queue is now empty, or if there was an error
+        if (eventQueue.size == 0 || !result.success) {
```

Comment:
should this trigger the error case of the callback if the events weren't fully flushed?

Reply:
basically in the case of close it might make sense for the close callback to error if it failed to flush every event

Reply:
yeah that makes sense